src/EnergyPlus/ScheduleManager.hh

```diff
@@ -328,6 +331,109 @@ namespace ScheduleManager {
+    };
+
+    std::istream &operator>>(std::istream &str, CSVRow &data);
+
+    void PreProcessIDF(EnergyPlus::EnergyPlusData &state, int &SchNum, int NumCommaFileSchedules);
+
+    struct schedInputIdfPreprocessObject
+    {
+        std::string name;     // <- can inherit from parent sched class
+        std::string fileName; // <- can inherit from parent sched class
+        int ScheduleTypePtr{-1};
+        int columnOfInterest{-1};
+        int numHourlyValues{-1};
+        int MinutesPerItem{-1};
+        int rowsToSkip{-1};
+        bool FileIntervalInterpolated{false};
+        std::basic_string<char> columnSeperator{"*"};
+        int columnarDataIndex{-1}; // TODO : Figure out what to do here (size_t to int conversion)
+        int SchNum{-1};
+
+        schedInputIdfPreprocessObject(std::string name,     // <- can inherit from parent sched class
+                                      std::string fileName, // <- can inherit from parent sched class
+                                      int ScheduleTypePtr = -1,
+                                      int columnOfInterest = -1,
+                                      int numHourlyValues = -1,
+                                      int MinutesPerItem = -1,
+                                      int rowsToSkip = -1,
+                                      bool FileIntervalInterpolated = false,
+                                      std::basic_string<char> columnSeperator = "*",
+                                      int SchNum = -1)
+        {
+
+            this->name = name;
+            this->fileName = fileName;
+            this->ScheduleTypePtr = ScheduleTypePtr;
+            this->columnOfInterest = columnOfInterest - 1; // - 1 to get index to match with c++ containers (start at 0)
+            this->numHourlyValues = numHourlyValues;
+            this->MinutesPerItem = MinutesPerItem;
+            this->rowsToSkip = rowsToSkip;
+            this->FileIntervalInterpolated = FileIntervalInterpolated;
+            this->columnSeperator = columnSeperator;
+            this->SchNum = SchNum;
+        }
+        void setColumnarDataIndex(int incomingData)
+        {
+            this->columnarDataIndex = incomingData;
+        }
+    };
+
+    void PopulateSetOfFilenames(const std::vector<schedInputIdfPreprocessObject> &allIdfSchedData, std::set<std::string> &setOfFilenames);
+
+    struct PreProcessedColumn
+    {
+        std::string name;
+        int columnOfInterest = -1;
+        int numHourlyValues = -1;
+        int MinutesPerItem = -1;
+        int rowsToSkip = -1;
+        bool FileIntervalInterpolated = false;
+        int ScheduleTypePtr = -1;
+        int SchNum = -1;
+        int rowCnt = 0;
+        int numerrors = 0;
+        std::string fileName;
+        std::string delimiter;
+        std::vector<Real64> vals; // rows to skip + number of hours*items/hour
+
+        PreProcessedColumn(std::string name, // <- can inherit from parent sched class
```

Comment:
We implement constructors here, so emplace_back can be called, which is better than push_back because creating a copy is avoided.